### PR TITLE
E2E : Networking.go Implement BeforeEach/AfterEach, and unique namespace...

### PR DESCRIPTION
**Pull request by jayunit100**
Follow up to #6600 , fixes a minor issue created in #6642 and is a better model to follow in the general cleanup we're doing for #6554

### CODE REVIEW

File: test/e2e/networking.go [hunk 1]
```diff
@@ -122,7 +78,54 @@ func LaunchNetTestPodPerNode(nodes *api.NodeList, name string, c *client.Client,
...
+		}
+
+		By("Creating a kubernetes client")
+		c, err = loadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Creating a namespace for this test suite")
+		_, err = c.Namespaces().Create(namespaceObj)
+		Expect(err).NotTo(HaveOccurred())
+		if err != nil {
```
Comment (satnam6502) [inline test/e2e/networking.go]: Can `err` ever be `nil` here?

File: test/e2e/networking.go [hunk 2]
```diff
@@ -122,7 +78,54 @@ func LaunchNetTestPodPerNode(nodes *api.NodeList, name string, c *client.Client,
...
+			Failf("Couldn't make ns %s", err)
+		}
+
+	})
+
+	AfterEach(func() {
+		By("Destroying namespace for this suite")
+		err := c.Namespaces().Delete(namespaceObj.ObjectMeta.Name)
+		if err != nil {
```
Comment (satnam6502) [inline test/e2e/networking.go]:
```
if err := c.Namespaces().Delete(namespaceObj.ObjectMeta.Name); err != nil {
...
}
```

File: test/e2e/networking.go (hunk 1 above)
Comment (jayunit100) [inline test/e2e/networking.go]: err definetly should always be nil else failure.

Comment (satnam6502) [inline test/e2e/networking.go]: If `err` is `nil` then it is already game over from the previous `Expect` line?

File: test/e2e/networking.go (hunk 2 above)
Comment (jayunit100) [inline test/e2e/networking.go]: yup ! implemented. thanks !

File: test/e2e/networking.go (hunk 1 above)
Comment (jayunit100) [inline test/e2e/networking.go]: ah yes true.  i need to scope it or rename.

File: test/e2e/networking.go [hunk 3]
```diff
@@ -122,7 +74,55 @@ func LaunchNetTestPodPerNode(nodes *api.NodeList, name string, c *client.Client,
...
+		Expect(err).NotTo(HaveOccurred())
+		if err != nil {
+			Failf("Couldn't make ns %s", err)
+		}
+
+	})
+
+	AfterEach(func() {
+		By("Destroying namespace for this suite")
+		if err := c.Namespaces().Delete(namespaceObj.ObjectMeta.Name); err != nil {
```
Comment (derekwaynecarr) [inline test/e2e/networking.go]: namespaceObj.Name

File: test/e2e/networking.go (hunk 1 above)
Comment (jayunit100) [inline test/e2e/networking.go]: i think the real problem is variable reuse... going to see if i can make cleaner

File: test/e2e/networking.go (hunk 3 above)
Comment (jayunit100) [inline test/e2e/networking.go]: something changed?

Comment (derekwaynecarr) [inline test/e2e/networking.go]: What you have is not incorrect, but I thought by convention in the project, we just did pod.Name, namespace.Name, etc.

File: test/e2e/networking.go (hunk 1 above)
Comment (jayunit100) [inline test/e2e/networking.go]: seems to be no good way to reuse err w/o shadowing cli ... :( not sure what the right way to reuse error variable is.  decalring a new one.

Comment (jayunit100) [inline test/e2e/networking.go]: @satnam6502 I guess one thing we can do is remove the if statement like you said,  i misinterpretted your "previous expect line" comment the first time, and went down the road of trying to scope err so its not reused between assertions, but i think thats unnecessary.  I'll do the simple thing and delete the if ***

File: test/e2e/networking.go (hunk 3 above)
Comment (jayunit100) [inline test/e2e/networking.go]: gotcha ill fix that too

**Comment by jayunit100** [comment]
okay @satnam6502   finished implementing fixes and tested.  

```
apache@localhost:~/kubernetes[networking-ns2]$ go run ./cmd/e2e/e2e.go --provider=local --host=http://127.0.0.1:8080 -t "Networking" --auth_config=/opt/kube_creds
Running Suite: Kubernetes e2e Suite run 1 of 1
==============================================
Random Seed: 1428615858 - Will randomize all specs
Will run 2 of 38 specs

STEP: Executing a successful http request from the external internet
STEP: Building a namespace api object
STEP: Creating a kubernetes client
>>> testContext.AuthConfig: /opt/kube_creds
STEP: Creating a namespace for this test suite
Networking 
  should function for intra-pod communication
  /home/apache/gopath/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/networking.go:277
[It] should function for intra-pod communication
  /home/apache/gopath/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/networking.go:277
STEP: Creating a service named [nettest] in namespace nettest-3791
STEP: Creating a webserver (pending) pod on each node
INFO: Creating pod nettest-0 on node 127.0.0.1
STEP: Waiting for the webserver pods to transition to Running state
STEP: waiting up to 5m0s for pod nettest-0 status to be running
INFO: Waiting for pod nettest-0 in namespace nettest-3791 status to be "running" (found "Pending") (1.976938ms)
STEP: Waiting for connectivity to be verified
INFO: About to make a proxy status call
INFO: Proxy status call returned in 1.443299ms
INFO: About to make a proxy status call
INFO: Proxy status call returned in 1.181766ms
INFO: About to make a proxy status call
INFO: Proxy status call returned in 1.815671ms
INFO: About to make a proxy status call
INFO: Proxy status call returned in 1.648408ms
INFO: About to make a proxy status call
INFO: Proxy status call returned in 1.533316ms
INFO: About to make a proxy status call
INFO: Proxy status call returned in 1.773058ms
INFO: About to make a proxy status call
INFO: Proxy status call returned in 1.48034ms
INFO: About to make a proxy status call
INFO: Proxy status call returned in 2.481438ms
INFO: Passed on attempt 7. Cleaning up.
STEP: Cleaning up the webserver pods
STEP: Cleaning up the service

• [SLOW TEST:21.045 seconds]
Networking
/home/apache/gopath/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/networking.go:279
  should function for intra-pod communication
  /home/apache/gopath/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/networking.go:277
------------------------------
SSSSSSSSSSSS
------------------------------
Networking 
  should provide unchanging, static URL paths for kubernetes api services.
  /home/apache/gopath/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/networking.go:146
[It] should provide unchanging, static URL paths for kubernetes api services.
  /home/apache/gopath/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/networking.go:146
STEP: testing: /validate
STEP: testing: /healthz
•SSSSSSSSSSSSSSSSSSSSSSSSSTEP: Destroying namespace for this suite

Ran 2 of 38 Specs in 21.223 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 36 Skipped I0409 17:44:39.716519   11148 driver.go:114] All tests pass
apache@localhost:~/kubernetes[networking-ns2]$ git log | head -3
commit 52befc19b91ab1489374578ded07e8384d740f9a
Author: jayunit100 <jay@apache.org>
Date:   Thu Apr 9 17:43:10 2015 -0400
```

**Comment by satnam6502** [comment]
I am still not sure why the external internet access test belongs in the before block but I don't feel very strongly about it. Copying @zmerlynn @ixdy @quinton-hoole who are actively involved in tests.

Thank you kindly for your contribution!

**Comment by jayunit100** [comment]
we can also moveach that call into a separate test.  Originally it was moved out of the tests because it's not a true test of kube networking per se and only is a test of the client

**Pull request merged**